azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionManager.java

```diff
@@ -325,7 +326,11 @@ else if (possibleLease.getOwner().compareTo(this.host.getHostName()) == 0)
                     	leasesOwnedByOthers.add(possibleLease);
                     }
             	}
-            	catch (ExecutionException e)
+        		// Most exceptions will arrive packaged as ExecutionException because they occur during a short-lived thread
+        		// down in AzureStorageCheckpointLeastManager. However, AzureBlobLease.isExpired calls Storage directly and
+        		// therefore can throw a plain StorageException. Handling is the same for all: log, notify, and move on to the
+            	// next partition.
+            	catch (ExecutionException|StorageException e)
             	{
             		this.host.logWithHost(Level.WARNING, "Failure getting/acquiring/renewing lease, skipping", e);
             		this.host.getEventProcessorOptions().notifyOfException(this.host.getHostName(), e, EventProcessorHostActionStrings.CHECKING_LEASES,
```

Comment:
>e [](start = 94, length = 1)

ExecutionException is our implementation detail and usually wraps the real exception that the Future actually threw.
if e instanceof ExecutionException pass on e.getCause().